systemtest/src/main/java/io/strimzi/systemtest/bases/IMetrics.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.bases;
+
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.strimzi.api.kafka.model.KafkaConnectResources;
+import io.strimzi.api.kafka.model.KafkaExporterResources;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.test.executor.Exec;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.strimzi.test.BaseITST.cmdKubeClient;
+import static io.strimzi.test.BaseITST.kubeClient;
+
+public interface IMetrics {
```

Comment:
Defining these methods in an interface for the convenience of inheriting them is an antipattern (though it's not exactly a problem here in the tests). Better to have them as `static` methods in a `Util` class which you can statically import if you want to avoid needing the class-qualify the names.

Reply:
We are not for 100% sure in this if we will use `interfaces` or `utils` classes. We wanted some feedback from devs (and now we have it). We want to follow better and right approach since we have plan to do some refactor for most of the test classes and for `AbstractST`. Interfaces has one pro - we will see which kind of tests are in specific class -> here we can see from interface name, that we will need some metrics methods...but we can read it from test class name directly.

If you think `utils` classes is better approach, I will gladly change it in this PR. Maybe it's better option, cause from `AbstractST` and `KafkaST` we want to move few of methods which are quite lon. We want to follow best practices in our code.

Reply:
The thing about an interface is you're defining public API. Speaking hypothetically, a client of a class which implements the interface will be able to call the methods. These methods are really implementation details of the test so it's not really appropriate for them to be publicly visible. (IIRC in Java > 8 it's possible to define non-public interface methods which wouldn't have this problem). More generally, inheriting methods is a stronger kind of coupling than delegation or calling a `static` method. You want weak coupling where possible because it generally makes code more reusable and easier to refactor. In this case you can only call these methods on an instance of the interface, but anyone can call a `static` method. You also don't gain anything from being able to override the method, which both suggest a static method might be a better choice.

Speaking practically the only client of an ST is the test running framework (in this case Junit), which won't call these methods, so the point is somewhat moot. But it's worth making because if you get into these habits you'll soon be doing the same thing elsewhere, where is does start to make a difference.

Enough of my lecturing.